reactioncommerce/reaction
# chore: access media through devserver app

Resolves #4214  
Impact: **minor**  
Type: **chore**

## Changes
When running `devserver` app rather than full Meteor app, images can now be downloaded, and the `collections.Media` on resolver context has the media FileCollection instance.

The path is the same. Only port is different when running locally. For example:

Normal reaction: http://localhost:3000/assets/files/Media/Jtj3k7QWNALNhwo5d/thumbnail/FvpPJZk.png
Devserver: http://localhost:3030/assets/files/Media/Jtj3k7QWNALNhwo5d/thumbnail/FvpPJZk.png

### Technical details

There isn't much new code here. Much of `fileCollections.js` has been moved to `no-meteor/setUpFileCollections.js` and wrapped in a function, so that some dependencies such as the Mongo collection can be passed in one way within the Meteor app and another way in devserver.

There are now two different `Media` file collections when running within Meteor. `Media` is the same as before, while `NoMeteorMedia` export is a `MongoFileCollection`, which is similar but does not set up DDP methods or work in client code.

When running devserver, there is just one `Media` export and that is the `MongoFileCollection`.

Because temp store / upload handling currently relies on Meteor Mongo observes, that part is still only in the Meteor file. We will eventually handle that in a different way, if we need to support browser uploads in the devserver app.

## Breaking changes
None

## Testing
1. Use `docker-compose up -d` to start both apps.
1. Find a product image in normal Reaction app, and get it's URL.
1. Change the port to 3030 and verify that you can see the same image there.
1. Make sure that media in general still works correctly for admins (upload, publish, delete, reorder images)
